doc/devdoc/cppcomments.rst

```diff
@@ -0,0 +1,40 @@
+Rules for C++  code comments
+============================
+
+* C++ comments single line (max 2 lines if sentence is long): //
+* C++ comments multile line: ``/* * * * */``
+
+* avoid doxygen comments in .cpp files, move to header file
+* avoid duplicating code in comments; only include code(parameters, functions etc) that also have additional context, remove redundancies
+* include the variable name in functions in header file to match cpp file.
+
+
+Functions and classes
+---------------------
+
+doxygen style: ``/** * */``
+
+
+Variables
+---------
+
+
+doxygen comments single line (max 2 lines if sentence is long): //!
```

Comment:
At the beginning of sentences, I'd capitalize in keeping with the doxygen website.
```suggestion
Doxygen comments single line (max 2 lines if sentence is long): //!
```

Should there be backticks around //!, and similar below?

Reply:
yes